stats/stats.go

```diff
@@ -71,14 +72,31 @@ func GetStatisticsSync(tempPath string, repos []string, l *log.Logger) repositor
 	return resultStats
 }
 
-func getStatsFromRepo(repo, tempPath string, l *log.Logger) (repository.Statistics, error) {
+func getStatsFromRepo(repo, tempPath, provider string, l *log.Logger) (repository.Statistics, error) {
 	if _, err := os.Stat(tempPath); os.IsNotExist(err) {
 		os.MkdirAll(tempPath, os.ModePerm)
 	}
-	gh := github.Downloader{}
-	path, err := gh.Download(repo, tempPath, l)
-	if err != nil {
-		l.Error(err)
+
+	var path string
+	var err error
+
+	switch provider {
+	case "github":
+		gh := github.Downloader{}
+		path, err = gh.Download(repo, tempPath, l)
+		if err != nil {
+			l.Error(err)
+		}
+
+		break
+	case "gitlab":
+		gl := gitlab.Downloader{}
+		path, err = gl.Download(repo, tempPath, l)
+		if err != nil {
+			l.Error(err)
+		}
+
+		break
```

Comment:
Is it not easier to do like  

```go
var path string
var err error
d := github.Downloader{} 
if provider == "gitlab" {
    d = gitlab.Downloader{} 
}

path, err = d.Download(repo, tempPath, l)
if err != nil {
    l.Error(err)
}
```
? What do you think?

Reply:
It wouldn't work because the types are different, But I think that what we can do is to transform this Downloader struct into an interface, what do you think?

Reply:
I see. I think this interface was created with this intention but it needs to be updade. 
https://github.com/filhodanuvem/polyglot/blob/master/downloader.go

Reply:
Made the change

Reply:
Fixed!